src/include/migraphx/matcher.hpp

```diff
@@ -857,23 +857,69 @@ auto skip_broadcasts_converts(Ms... ms)
     return skip(name("broadcast", "multibroadcast", "contiguous", "convert"))(ms...);
 }
 
-template <class T>
-inline auto has_value(T x, float tolerance = 1e-6)
+template <class F>
+inline auto literal_value_checker(F f)
 {
     return skip_broadcasts_converts(make_basic_pred_matcher([=](instruction_ref ins) {
         if(ins->name() != "@literal")
             return false;
         auto l = ins->get_literal();
         if(l.empty())
             return false;
+        return f(l);
+    }));
+}
+
+/**
+ * Uses integer multiples of the corresponding floating point epsilon and
+ * compares with abs(y - x) < eps * (atol_mult + rtol_mult * abs(y))
+ */
+template <class T>
+inline auto has_value(T x, std::size_t atol_mult, std::size_t rtol_mult)
+{
+    return literal_value_checker([=](migraphx::literal l) {
         bool b = false;
         l.visit([&](auto v) {
-            if(std::all_of(
-                   v.begin(), v.end(), [&](auto val) { return std::fabs(val - x) < tolerance; }))
+            // cast to the literal's data type before comparing
+            using type = typename decltype(v)::value_type;
+            auto eps   = std::numeric_limits<type>::epsilon();
+            if(std::all_of(v.begin(), v.end(), [&](auto val) {
+                   return std::fabs(val - static_cast<type>(x)) <
+                          eps * (atol_mult + rtol_mult * std::fabs(val));
+               }))
+            {
                 b = true;
+            }
         });
         return b;
-    }));
+    });
+}
+
+/// Check literal value with an explicit floating point tolerance
+template <class T, class F, MIGRAPHX_REQUIRES(std::is_floating_point<F>{})>
+inline auto has_value(T x, F tolerance)
```

Comment:
I think this should be removed.

Reply:
We can remove this, but what would be the desired functionality of previous calls that used a fixed tolerance?

Reply:
This seems like a much better implementation than the older tolerance version. We never used the tolerance parameter except for gelu, most likely because of fp16 which this implementation should handle better.